### PR TITLE
go-ios: init at 1.0.115

### PR DESCRIPTION
**Pull request by eyJhb**
###### Description of changes
Package for running various commands on ios devices, written in Golang - https://github.com/danielpaulus/go-ios .

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/go-ios/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  vendorSha256 = "sha256-GtXnjK3tAEIj4l0HeXYjdJ4MWq824euKkfiy/vA76Sw=";
+
+  postPatch = ''
+    # do not build this (does not work)
+    rm -rf restapi
+
+    # remove these tests, as they fail
+    find . -name '*_integration_test.go' -delete
+    rm ios/testmanagerd/xcuitestrunner_test.go
+  '';
```
Comment (emilylange) [inline pkgs/tools/misc/go-ios/default.nix:28]:
```suggestion
  excludedPackages = [
    "restapi"
  ];

  checkFlags = [
    "-tags=fast"
  ];
```

Upstream flags those integration/failing tests with build tags, which we can re-use :)

File: pkgs/tools/misc/go-ios/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  postInstall = ''
+    # aligns the binary with what is expected from go-ios
+    mv $out/bin/go-ios $out/bin/ios
+  '';
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = with lib; {
+    description = "This is an operating system independent implementation of iOS device features. You can run UI tests, launch or kill apps, install apps etc. with it.";
```
Comment (emilylange) [inline pkgs/tools/misc/go-ios/default.nix:38]: E.g.

```suggestion
    description = "An operating system independent implementation of iOS device features";
```

See https://nixos.org/manual/nixpkgs/unstable/#var-meta-description and https://github.com/NixOS/nixpkgs/blob/0138c0f6edc6cc5cdc2f3894a1adc3e0ace4ce1a/CONTRIBUTING.md#submitting-changes

File: pkgs/tools/misc/go-ios/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
+{
+  lib
```
Comment (emilylange) [inline pkgs/tools/misc/go-ios/default.nix:2]:
```suggestion
{ lib
```

Formatting nit

**Review by emilylange — CHANGES_REQUESTED** [review]

**Comment by eyJhb** [comment]
Should all be fixed @emilylange ! Thanks ! :)

**Review by emilylange — CHANGES_REQUESTED** [review]
Hash needs to be updated. Sorry^^

**Pull request merged**